foodsaving/userauth/models.py

```diff
@@ -0,0 +1,57 @@
+from datetime import timedelta
+
+from django.db.models import CharField, ForeignKey, Manager
+from django.db import models
+from django.conf import settings
+from django.utils import crypto, timezone
+
+from foodsaving.base.base_models import BaseModel
+
+# TODO: Ensure each user can have at most one verification code of a certain type at a time.
```

Comment:
Seems good, you can do that with a one-liner by adding a `unique_together` constraint. I used it in the invitations code, for example.